src/geometry/euclidean.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Joseph Myers.
+-/
+import algebra.big_operators
+import analysis.normed_space.real_inner_product
+import analysis.normed_space.add_torsor
+
+noncomputable theory
+open_locale big_operators
+
+/-!
+# Euclidean spaces
+
+This file defines Euclidean affine spaces.
+
+## Implementation notes
+
+Rather than requiring Euclidean affine spaces to be finite-dimensional
+(as in the definition on Wikipedia), this is specified only for those
+theorems that need it.
+
+## References
+
+* https://en.wikipedia.org/wiki/Euclidean_space
+
+-/
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
+/-- A `euclidean_affine_space V P` is an affine space with points `P`
+over an `inner_product_space V`. -/
+class euclidean_affine_space (V : Type*) (P : Type*) [inner_product_space V] [metric_space P]
```

Comment:
This definition does not brin anything new, since you are extending without adding new fields. This means that it is not needed: you could delete it, and in applications just mention the type classes `[inner_product_space V] [metric_space P] [normed_add_torsor V P]`.

If you want to keep the name, you can do it, but it should just be an abbreviation for `normed_add_torsor`. You can have a look at the "definition" of vector spaces, which are just an abbreviation for modules.

Reply:
Changed to use `abbreviation`.